src/basic/utf8.h

```diff
@@ -28,6 +28,7 @@
 #include "missing.h"
 
 #define UTF8_REPLACEMENT_CHARACTER "\xef\xbf\xbd"
+#define UTF8_BOM "\xef\xbb\xbf"
```

Comment:
Such acronyms leave room for confusion. Can we call this `UTF8_BYTE_ORDER_MARK` instead?